libs/server/auth/infrastructure/src/lib/guards/auth.guard.ts

```diff
@@ -0,0 +1,31 @@
+import { CanActivate, ExecutionContext, Injectable } from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import { Request, Response } from 'express';
+
+@Injectable()
+export class AuthGuard implements CanActivate {
+  constructor(private jwtService: JwtService) {}
+
+  async canActivate(context: ExecutionContext): Promise<boolean> {
+    const httpContext = context.switchToHttp();
+    const req: Request = httpContext.getRequest();
+    const authHeader = req.header('Authorization');
+    const jwtToken = authHeader && authHeader.split(' ')[1];
+
+    if (!jwtToken) {
+      return false;
```

Comment:
shouldn't it throw an UnathorizedError?

Reply:
![image](https://user-images.githubusercontent.com/24508253/77151574-822f5480-6a96-11ea-801e-b2ec9038162d.png)